crates/red_knot_python_semantic/src/semantic_index/use_def/symbol_state.rs

```diff
@@ -173,11 +171,46 @@ impl SymbolDeclarations {
             visibility_constraints: self.visibility_constraints.iter(),
         }
     }
+
+    fn merge(&mut self, b: Self, visibility_constraints: &mut VisibilityConstraints) {
+        let mut a = Self::default();
+        std::mem::swap(&mut a, self);
+        self.live_declarations = a.live_declarations.clone();
+        self.live_declarations.union(&b.live_declarations);
+        let a = izip!(
+            a.live_declarations.iter(),
+            a.visibility_constraints.into_iter()
+        );
+        let b = izip!(
+            b.live_declarations.iter(),
+            b.visibility_constraints.into_iter()
+        );
+        for zipped in a.merge_join_by(b, |(a_decl, _), (b_decl, _)| a_decl.cmp(b_decl)) {
+            match zipped {
+                EitherOrBoth::Both((_, a_vis_constraint), (_, b_vis_constraint)) => {
+                    let vis_constraint = visibility_constraints
+                        .add_or_constraint(a_vis_constraint, b_vis_constraint);
+                    self.add_via_merge(vis_constraint);
+                }
+
+                EitherOrBoth::Left((_, vis_constraint))
+                | EitherOrBoth::Right((_, vis_constraint)) => {
+                    self.add_via_merge(vis_constraint);
+                }
+            }
+        }
+    }
+
+    fn add_via_merge(&mut self, vis_constraint: ScopedVisibilityConstraintId) {
+        // SAFETY: This can only be called from `merge`, since it assumes that we are adding
+        // elements in order, sorted by `decl`.
+        self.visibility_constraints.push(vis_constraint);
```

Comment:
Should we inline this function because there's nothing that actually guarantees the constraint of the safety comment -- the function can be called from anywhere in this module.